File: Problems/LeetCode/2038.Remove-Colored-Pieces-if-Both-Neighbors-are-the-Same-Color.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int moves[2] = {0, 0};
        for(int i = 1; i < colors.size() - 1; ++i) {
            moves[colors[i] - 'A'] += colors[i] == colors[i - 1] && colors[i] == colors[i + 1];
        }
        return moves[0] > moves[1];
    }
};

File: Problems/LeetCode/2038.Remove-Colored-Pieces-if-Both-Neighbors-are-the-Same-Color.py
# https://leetcode.com/problems/remove-colored-pieces-if-both-neighbors-are-the-same-color/

class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        moves = dict(A=0, B=0)
        for i in range(1, len(colors) - 1):
            moves[colors[i]] += colors[i - 1] == colors[i] == colors[i + 1]
        return moves["A"] > moves["B"]
